velox/functions/prestosql/ArrayDuplicates.cpp

```diff
@@ -0,0 +1,204 @@
+        if (elements->isNullAt(i)) {
+          numOfNulls++;
+          if (numOfNulls == 2) {
+            rawIndices[indexCursor] = i;
+            indexCursor++;
+          }
+        } else {
+          T value = elements->valueAt<T>(i);
+          auto it = uniqueMap.find(value);
+          if (it == uniqueMap.end()) {
+            uniqueMap[value] = true;
+          } else if (it->second) {
+            it->second = false;
+            rawIndices[indexCursor] = i;
+            indexCursor++;
+          }
+        }
+      }
+
+      uniqueMap.clear();
+      rawSizes[row] = indexCursor - rawOffsets[row];
+
+      std::sort(
+          rawIndices + rawOffsets[row],
+          rawIndices + indexCursor,
+          lib::Index2ValueNullableLess<T>(elements));
+    });
+
+    newIndices->setSize(indexCursor * sizeof(vector_size_t));
+    auto newElements =
+        BaseVector::transpose(newIndices, std::move(elementsVector));
+
+    // Prepare and return result set.
+    auto resultArray = std::make_shared<ArrayVector>(
+        pool,
+        outputType,
+        nullptr,
+        numOfRows,
+        std::move(newOffsets),
+        std::move(newSizes),
+        std::move(newElements),
+        0);
+
+    context->moveOrCopyResult(resultArray, rows, result);
+  }
+
+ private:
+  // Validate element type inside the vector
+  static void validateElementType(const TypeKind& kind) {
+    VELOX_USER_CHECK(
+        kind == TypeKind::BIGINT || kind == TypeKind::VARCHAR,
+        "array_duplicates only takes argument of type array(bigint/varchar).");
+  }
+};
+
+// Validate number of parameters and types.
+void validateType(const std::vector<exec::VectorFunctionArg>& inputArgs) {
+  VELOX_USER_CHECK_EQ(
+      inputArgs.size(), 1, "array_duplicates requires exactly one parameter");
+
+  auto arrayType = inputArgs.front().type;
+  VELOX_USER_CHECK_EQ(
+      arrayType->kind(),
+      TypeKind::ARRAY,
+      "array_duplicates requires arguments of type ARRAY");
+}
+
+// Create function template based on type.
+template <TypeKind kind>
+std::shared_ptr<exec::VectorFunction> createTyped(
+    const std::vector<exec::VectorFunctionArg>& inputArgs) {
+  VELOX_CHECK_EQ(inputArgs.size(), 1);
+
+  using T = typename TypeTraits<kind>::NativeType;
+  return std::make_shared<ArrayDuplicatesFunction<T>>();
+}
+
+// Create function.
+std::shared_ptr<exec::VectorFunction> create(
+    const std::string& /* name */,
+    const std::vector<exec::VectorFunctionArg>& inputArgs) {
+  validateType(inputArgs);
+  auto elementType = inputArgs.front().type->childAt(0);
+
+  return VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
+      createTyped, elementType->kind(), inputArgs);
+}
+
+// Define function signature.
+// array(T) -> array(T) where T must be coercible to bigint or varchar.
+std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
+  static const std::vector<std::string> kSupportedTypes = {"bigint", "varchar"};
```

Comment:
nit: I think it would be easier to read if the signatures are just hard-coded

```
return {
exec::FunctionSignatureBuilder()
            .returnType("array(bigint)")
            .argumentType("array(bigint)")
            .build(),
exec::FunctionSignatureBuilder()
            .returnType("array(varchar)")
            .argumentType("array(varchar)")
            .build()
};
```